specification/app/resource-manager/Microsoft.App/preview/2023-11-02-preview/examples/AppSpaces_CreateOrUpdate.json

```diff
@@ -0,0 +1,43 @@
+{
+  "parameters": {
+    "subscriptionId": "34adfa4f-cedf-4dc0-ba29-b6d1a69ab345",
+    "resourceGroupName": "rg",
+    "appSpaceName": "testappSpace0",
+    "api-version": "2023-11-02-preview",
+    "appSpaceEnvelope": {
+      "location": "East US",
+      "properties": {
+        "friendlyName": "Test App Space 0",
+        "primaryResourceId": ""
+      }
+    }
+  },
+  "responses": {
+    "200": {
+      "headers": {},
+      "body": {
+        "id": "/subscriptions/34adfa4f-cedf-4dc0-ba29-b6d1a69ab345/resourceGroups/rg/providers/Microsoft.App/appSpaces/testappSpace0",
+        "name": "testappSpace0",
+        "type": "Microsoft.App/appSpaces",
+        "location": "East US",
+        "properties": {
+          "friendlyName": "Test App Space 0",
+          "primaryResourceId": ""
+        }
```

Comment:
again here what is this primary resource id used for ? How is it different from the main id of the resource ?

Reply:
An App Space is a grouping of resources, which we call app space resources. Each app space resource tracks a particular resource, be it a containerapp in the Microsoft.App RP, or something in another RP. The primary resource ID is the resource ID of the primary app space resource in the app space. The customer doesn't have to have a primary resource, but if they do, portal uses this field to display the primary resource first.